src/core/models.py

```diff
@@ -142,13 +142,15 @@ class ODServer(StateBase):
 
     def __init__(
         self,
+        model: Model,
```

Comment:
Please, let's try to avoid doing this if possible...

Let me highlight regarding the structure of "new-style" charms.
There has been a strong Separation of Concerns applied, typically to avoid "almighty" components such as `model` or `charm` wired in spaces where they aren't neeed. 
The `src/core/models` module implements the representation of Charm Components (such as `Unit`, `App`, etc.) based on Relation Data or very basic additional, independent actions (such as `socket` calls).

To my understanding, binding information is outside of the scope of the `ODServer` object, as it only depends on the `ClusterState` object (implementing the `peer_relation` property). Thus I rather suggest a `peer_binding` in `src/core/cluster` (see example on [other "new-style" charm](https://github.com/canonical/kafka-operator/blob/7cfebeee4066b0258dde0560a70b9c68c7ebca7a/src/core/cluster.py#L296-L303
) ).

Now a next quesetion is if we want to use the new `osd_charm.state.bind_address` in all places where we've been using `private_ip` before? If yes, we may want to apply corresponding changes on `events/tls` and `managers/tls`. Otherwise changing `managers/config` is enough.